teachertool/src/components/CatalogOverlay.tsx

```diff
@@ -106,34 +127,72 @@ const CatalogList: React.FC = () => {
         announceToScreenReader(lf("Added '{0}' to checklist.", getReadableCriteriaTemplate(c)));
     }
 
+    function getItemIdForTag(tag: string) {
+        return `accordion-item-${tag}`;
+    }
+
+    function onTagExpandToggled(tag: string, expanded: boolean) {
+        if (expanded) {
+            /* await */ addExandedCatalogTagAsync(tag);
+        } else {
+            /* await */ removeExpandedCatalogTagAsync(tag);
+        }
+    }
+
+    const tags = Object.keys(criteriaGroupedByTag);
+    if (tags.length === 0) {
+        return null;
+    }
+
+    let expandedTags = getExpandedCatalogTags();
+    if (!expandedTags) {
+        // If we haven't saved an expanded set, default expand the first one.
+        addExandedCatalogTagAsync(tags[0]);
+        expandedTags = [tags[0]];
+    }
+
+    const expandedIds = expandedTags.map(t => getItemIdForTag(t));
     return (
-        <div className={css["catalog-list"]}>
-            {criteria.map(c => {
-                const existingInstanceCount = teacherTool.checklist.criteria.filter(
-                    i => i.catalogCriteriaId === c.id
-                ).length;
-                const isMaxed = c.maxCount !== undefined && existingInstanceCount >= c.maxCount;
+        <Accordion className={css["catalog-list"]} multiExpand={true} defaultExpandedIds={expandedIds}>
+            {tags.map(tag => {
                 return (
-                    c.template && (
-                        <Button
-                            id={`criteria_${c.id}`}
-                            title={getReadableCriteriaTemplate(c)}
-                            key={c.id}
-                            className={css["catalog-item"]}
-                            label={
-                                <CatalogItemLabel
-                                    catalogCriteria={c}
-                                    isMaxed={isMaxed}
-                                    recentlyAdded={recentlyAddedIds[c.id] !== undefined}
-                                />
-                            }
-                            onClick={() => onItemClicked(c)}
-                            disabled={isMaxed}
-                        />
-                    )
+                    <Accordion.Item
+                        itemId={getItemIdForTag(tag)}
+                        onExpandToggled={expanded => onTagExpandToggled(tag, expanded)}
+                        key={getItemIdForTag(tag)}
+                    >
+                        <Accordion.Header>{tag}</Accordion.Header>
```

Comment:
Should the tag get `lf`-ed here?

Reply:
It should be translated before we get here. We actually can't lf variables, as I recal.

Reply:
Ah, yes, you're totally right.

Reply:
there's an rlf function for variables, which does a runtime lookup for the translation. not necessary if already translated at this point, but FYI.

this is also why you sometimes see comments like this:

```
// DO NOT REMOVE!
// lf("Loops")
```

it's just to make it so that "Loops" get's added to the json file for lookup later with rlf